Source/EasyNetQ/Topology/Queue.cs

```diff
@@ -1,68 +1,56 @@
 ﻿using System.Collections.Generic;
-using System.Linq;
+using System.Collections.ObjectModel;
 
 namespace EasyNetQ.Topology
 {
-    /// <inheritdoc />
-    public sealed class Queue : IQueue
+    /// <summary>
+    ///     Represents an AMQP queue
+    /// </summary>
+    public readonly struct Queue
```

Comment:
👍

Reply:
as we expected that this is used, the struct it's a good choice here, but take in mind that on every assignment then all the struct is deep-copied, so also when we pass it to a method (if we don't use ref), are we really ok with this? queue contains strings that are reference and are cached btw, so it should not be a great deal, it should also benefit the GC if we don't envelop it in a class or do strange thing in the struct, so basically I'm ok with this change but with a little bit of a warn.

Reply:
> on every assignment then all the struct is deep-copied, so also when we pass it to a method (if we don't use ref), are we really ok with this? 

`readonly struct` almost always go with `in` modifier for its method arguments